Cannot run `just tauri-build` but `just dev` is fine

I noticed you introduced `just` as the recommended build tool.
I pulled the latest code, ran `just setup`, then `just tauri-build` but it failed with 
```
just tauri-build
tauri build --target universal-apple-darwin
        Info Looking up installed tauri packages to check mismatched versions...
     Running beforeBuildCommand `just frontend-build`
node scripts/sync-version.cjs
[sync-version] package.json 버전: 1.1.0
[sync-version] ✓ Cargo.toml → 1.1.0
[sync-version] ✓ tauri.conf.json → 1.1.0 (이전: 1.1.0)
[sync-version] 모든 파일이 1.1.0로 동기화되었습니다.
pnpm exec tsc --build .
src/hooks/useNativeUpdater.ts:1:44 - error TS6133: 'useRef' is declared but its value is never read.

1 import { useState, useEffect, useCallback, useRef } from "react";
                                             ~~~~~~

src/store/useAppStore.ts:68:26 - error TS2554: Expected 2 arguments, but got 3.

68   ...createMetadataSlice(...args),
                            ~~~~~~~


Found 2 errors.

error: Recipe `frontend-build` failed on line 65 with exit code 2
beforeBuildCommand `just frontend-build` failed with exit code 2
       Error beforeBuildCommand `just frontend-build` failed with exit code 2
error: Recipe `tauri-build` failed on line 76 with exit code 1
```

I can run `just dev` but this won't generate the DMG

--- a/src/components/MessageViewer/helpers/heightEstimation.ts
+++ b/src/components/MessageViewer/helpers/heightEstimation.ts
@@ -26,6 +26,11 @@ const HEIGHT_DEFAULTS = {
  * This is used as the initial estimate before actual measurement.
  */
 export function estimateMessageHeight(item: FlattenedMessage): number {
+  // Hidden placeholder has fixed height
+  if (item.type === "hidden-placeholder") {
+    return 40; // Compact placeholder height
+  }
+
   const { message, isGroupMember, isProgressGroupMember, agentTaskGroup, agentProgressGroup } = item;
 
   // Group members are hidden (height: 0)

--- a/src/components/MessageViewer/hooks/useMessageVirtualization.ts
+++ b/src/components/MessageViewer/hooks/useMessageVirtualization.ts
@@ -135,7 +135,7 @@ export const useMessageVirtualization = ({
       if (directIndex !== undefined) {
         const item = flattenedMessages[directIndex];
         // If this is a group member, find the leader instead
-        if (item?.isGroupMember || item?.isProgressGroupMember) {
+        if (item?.type === "message" && (item.isGroupMember || item.isProgressGroupMember)) {
           const leaderIndex = findGroupLeaderIndex(
             uuid,
             flattenedMessages,

--- a/src/components/ProjectTree.tsx
+++ b/src/components/ProjectTree.tsx
@@ -54,7 +54,7 @@ interface ProjectTreeProps {
 export const ProjectTree: React.FC<ProjectTreeProps> = ({
   projects,
   sessions,
-  selectedProject,
+  selectedProject: _selectedProject,
   selectedSession,
   onProjectSelect,
   onSessionSelect,
@@ -360,7 +360,6 @@ export const ProjectTree: React.FC<ProjectTreeProps> = ({
                     <div className="ml-4 pl-3 border-l-2 border-blue-500/20 space-y-0.5">
                       {group.projects.map((project) => {
                         const isProjectExp = isProjectExpanded(project.path);
-                        const isSelected = selectedProject?.path === project.path;
 
                         return (
                           <div key={project.path}>
@@ -550,7 +549,6 @@ export const ProjectTree: React.FC<ProjectTreeProps> = ({
                       {allGroupProjects.map((project, idx) => {
                         const isProjectExp = isProjectExpanded(project.path);
                         const isMain = idx === 0;
-                        const isSelected = selectedProject?.path === project.path;
 
                         return (
                           <div key={project.path}>
@@ -678,7 +676,6 @@ export const ProjectTree: React.FC<ProjectTreeProps> = ({
             {/* Note: directory mode shows ALL projects in groups, so no ungrouped section */}
             {(groupingMode === "none" ? projects : groupingMode === "worktree" ? displayProjects : []).map((project) => {
               const isExpanded = isProjectExpanded(project.path);
-              const isSelected = selectedProject?.path === project.path;
 
               return (
                 <div key={project.path}>

--- a/src/store/slices/metadataSlice.ts
+++ b/src/store/slices/metadataSlice.ts
@@ -5,6 +5,7 @@
  */
 
 import { invoke } from "@tauri-apps/api/core";
+import type { StateCreator } from "zustand";
 import type {
   UserMetadata,
   SessionMetadata,
@@ -13,7 +14,7 @@ import type {
 } from "../../types";
 import { DEFAULT_USER_METADATA } from "../../types";
 import { matchGlobPattern } from "../../utils/globUtils";
-import type { StoreSet, StoreGet, FullAppStore } from "./types";
+import type { FullAppStore } from "./types";
 
 // ============================================================================
 // State Interface
@@ -87,10 +88,12 @@ export const initialMetadataState: MetadataSliceState = {
 // Slice Creator
 // ============================================================================
 
-export const createMetadataSlice = (
-  set: StoreSet<FullAppStore>,
-  get: StoreGet<FullAppStore>
-): MetadataSlice => ({
+export const createMetadataSlice: StateCreator<
+  FullAppStore,
+  [],
+  [],
+  MetadataSlice
+> = (set, get) => ({
   ...initialMetadataState,
 
   loadMetadata: async () => {

--- a/src/utils/worktreeUtils.ts
+++ b/src/utils/worktreeUtils.ts
@@ -364,11 +364,11 @@ export function toDisplayPath(fullPath: string, homePath?: string): string {
 function detectHomePath(path: string): string | null {
   // macOS: /Users/username/...
   const macMatch = path.match(/^(\/Users\/[^/]+)/);
-  if (macMatch) return macMatch[1];
+  if (macMatch?.[1]) return macMatch[1];
 
   // Linux: /home/username/...
   const linuxMatch = path.match(/^(\/home\/[^/]+)/);
-  if (linuxMatch) return linuxMatch[1];
+  if (linuxMatch?.[1]) return linuxMatch[1];
 
   return null;
 }

